pcre_wrapper.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2018 Couchbase, Inc. All rights reserved.
+
+// +build pcre all
+
+package gojsonsm
+
+import (
+	"errors"
+	"github.com/glenn-brown/golang-pkg-pcre/src/pkg/pcre"
+)
+
+type PcreWrapper struct {
+	pcreRegex *pcre.Regexp
+}
+
+func (wrapper *PcreWrapper) IsSupported() error {
+	return nil
+}
+
+func MakePcreCheckWrapper() PcreWrapperInterface {
```

Comment:
I don't see where this is used later, but why not just wait until the first expression using PCRE expressions is compiled before we generate this error?  This is particularly necessary to ensure the library still functions without PCRE.

Reply:
I'm not sure what you mean by "generate this error". Could you please clarify? The line of code you've highlighted just returns a struct to SimpleParser. 
If users don't specify the pcre build flag, we will use the empty (disabled as you like to call it) wrapper and that will allow non-pcre things to work. Any PCRE related things called will return an error.

Reply:
I suspect you can remove `IsSupported` based on the same logic as well.

Reply:
Thanks. Makes sense.